src/common/types/course.types.ts

```diff
@@ -0,0 +1,38 @@
+export interface ICourse {
```

Comment:
If you use the `I` prefix, use it for all interfaces, not only for one... Use another name for the interface if the Course already exists.